project-base/storefront/helpers/pagination/loadMore.ts

```diff
@@ -0,0 +1,75 @@
+import { getEndCursor } from 'components/Blocks/Product/Filter/helpers/getEndCursor';
+import {
+    BrandProductsQueryApi,
+    CategoryProductsQueryApi,
+    FlagProductsQueryApi,
+    ListedProductConnectionFragmentApi,
+    Maybe,
+    ProductFilterApi,
+    ProductOrderingModeEnumApi,
+} from 'graphql/generated';
+import { Client } from 'urql';
+
+export const mergeNewProductsWithPreviouslyQueried = (
+    newProductsData: BrandProductsQueryApi | CategoryProductsQueryApi | FlagProductsQueryApi,
+    previouslyQueriedProductsFromCache:
+        | BrandProductsQueryApi
+        | CategoryProductsQueryApi
+        | FlagProductsQueryApi
+        | undefined,
+) => {
+    newProductsData.products = {
+        ...newProductsData.products,
+        edges: mergeProductEdges(previouslyQueriedProductsFromCache?.products.edges, newProductsData.products.edges),
+    };
+
+    return newProductsData;
+};
```

Comment:
You shouldn't modify existing data from outside of the function. But I have bigger idea, what about to cut `edges` here and simply return array of edges content? Since we don't need anything else besides `edges`, it would make sense to me.

Then we would have instead of this 
![image](https://github.com/shopsys/shopsys/assets/11465991/e8bdc675-2417-4387-a773-4e0ca7bd5505)

this
![image](https://github.com/shopsys/shopsys/assets/11465991/826da5c4-8f07-41ad-81cd-25fdb21c46b3)

and you can remove this function

Reply:
We discussed this on a call and I changed the behavior in https://github.com/shopsys/shopsys/pull/2695/commits/4665f68195be1d49c3197e2c4c2b249820388f40

Reply:
Cool! I think it was worth the time. Thanks for your effort! 🙏